src/main/java/org/dependencytrack/tasks/OsvDownloadTask.java

```diff
@@ -297,4 +310,28 @@ private boolean isVulnerabilitySourceClashingWithGithubOrNvd(String source) {
         return Vulnerability.Source.GITHUB.toString().equals(source)
                 || Vulnerability.Source.NVD.toString().equals(source);
     }
+
+    public static List<String> getEcosystems() {
+        ArrayList<String> ecosystems = new ArrayList<>();
+        String url = OSV_BASE_URL + "ecosystems.txt";
+        HttpUriRequest request = new HttpGet(url);
+        try (final CloseableHttpResponse response = HttpClientPool.getClient().execute(request)) {
+            final StatusLine status = response.getStatusLine();
+            if (status.getStatusCode() == 200) {
+                try (InputStream in = response.getEntity().getContent();
+                     Scanner scanner = new Scanner(in, StandardCharsets.UTF_8.name())) {
+                    while (scanner.hasNextLine()) {
+                        if(!scanner.nextLine().isBlank()) {
+                            ecosystems.add(scanner.nextLine().trim());
```

Comment:
The method currently does not return all available ecosystems, but instead:

```
["DWF","Debian:10","Debian:3.0","Debian:4.0","Debian:6.0","Debian:8","GSD","Go","JavaScript","Maven","OSS-Fuzz","Pub","RubyGems","crates.io"]
```

Further, an exception is logged every time it's called:

```
2022-09-06 19:03:42,890 ERROR [OsvDownloadTask] Exception while executing Http request for ecosystems
java.util.NoSuchElementException: No line found
	at java.base/java.util.Scanner.nextLine(Scanner.java:1651)
	at org.dependencytrack.tasks.OsvDownloadTask.getEcosystems(OsvDownloadTask.java:325)
	at org.dependencytrack.resources.v1.OsvEcosytemResource.getAllEcosystems(OsvEcosytemResource.java:54)
```

The issue is that `scanner.nextLine()` advances the underlying "pointer" on each invocation, so you're always skipping a line here, and in the end you're trying to advance the scanner beyond the last line.

```suggestion
                        final String line = scanner.nextLine();
                        if(!line.isBlank()) {
                            ecosystems.add(line.trim());
```